src/ServiceControl.Config/Commands/ForceUpgradeServiceControlInstanceCommand.cs

```diff
@@ -0,0 +1,129 @@
+﻿namespace ServiceControl.Config.Commands;
+
+using System.Linq;
+using System.Threading.Tasks;
+using Caliburn.Micro;
+using Framework.Commands;
+using Events;
+using Framework.Modules;
+using Framework;
+using ServiceControlInstaller.Engine.Configuration.ServiceControl;
+using ServiceControlInstaller.Engine.FileSystem;
+using ServiceControlInstaller.Engine.Instances;
+using ServiceControlInstaller.Engine.ReportCard;
+using ServiceControlInstaller.Engine.Validation;
+using UI.AdvancedOptions;
+
+class ForceUpgradeServiceControlInstanceCommand : AwaitableAbstractCommand<ServiceControlAdvancedViewModel>
+{
+    public ForceUpgradeServiceControlInstanceCommand(
+        IServiceControlWindowManager windowManager,
+        IEventAggregator eventAggregator,
+        ServiceControlInstanceInstaller serviceControlInstaller) : base(ForcedUpgradeAllowed)
+    {
+        this.windowManager = windowManager;
+        this.eventAggregator = eventAggregator;
+        this.serviceControlInstaller = serviceControlInstaller;
+    }
+
+    [FeatureToggle(Feature.LicenseChecks)]
+    public bool LicenseChecks { get; set; }
+
+    static bool ForcedUpgradeAllowed(ServiceControlAdvancedViewModel model)
+    {
+        var instance = InstanceFinder.ServiceControlInstances().FirstOrDefault(i => i.Name == model.Name);
+
+        //HINT: Force upgrade is available only primary v4 instance, running on RavenDB 3.5
+        return instance != null && instance.Version.Major == 4 && instance.PersistenceManifest.Name == StorageEngineNames.RavenDB35;
+    }
+    public override async Task ExecuteAsync(ServiceControlAdvancedViewModel model)
+    {
+        if (LicenseChecks)
+        {
+            var licenseCheckResult = serviceControlInstaller.CheckLicenseIsValid();
+            if (!licenseCheckResult.Valid)
+            {
+                await windowManager.ShowMessage("LICENSE ERROR", $"Upgrade could not continue due to an issue with the current license. {licenseCheckResult.Message}.  Contact contact@particular.net", hideCancel: true);
+                return;
+            }
+        }
+
+        if (ForcedUpgradeAllowed(model) == false)
```

Comment:
Either have all if's use the `== false` or 'false ==' style or use `!` but now its mixed.
```suggestion
        if (!ForcedUpgradeAllowed(model))
```